### ISSUE TITLE
Update ubuntu dependencies

### ISSUE THREAD

**Issue by grummerd** [monero-project/kovri#310]
**By submitting this issue, I confirm the following:**

- I have read and understood the [contributor guide](https://github.com/monero-project/kovri/blob/master/doc/CONTRIBUTING.md).
- I have checked that the issue I am reporting can be replicated or that the feature I am suggesting is not present.
- I have checked opened or recently closed [pull requests](https://github.com/monero-project/kovri/pulls) for existing solutions/implementations to my issue/suggestion.

*Place an X inside the bracket to confirm*
- [ x ] I confirm.

---

Recently tried unsuccessfully to install kovri. I'm using Ubuntu 16.04.2 LTS

HOW TO REPRODUCE
--------------------------------

Get a clean system running Ubuntu 16.04.2 64bit version

Build using 

> make all-options

instead of 

> make dependencies

From the docs, i wanted UPnP, so `make all-options` seemed like the correct choice instead of `make dependencies`

If `make all-options` is an acceptable option then please update the docs, [install dependencies](https://github.com/monero-project/kovri/blob/master/doc/BUILDING.md)

What to fix
-------------------

1. Update the dependencies

>  sudo apt-get install libcrypto++dev libcppnetlib-dev

1. Downgrade requirement

libcppnetlib-dev in ubuntu apt repository is v0.11.2+dfsg1-2
Kovri requires libcppnetlib-dev v0.12.0-final

Is it possible to downgrade the requirement to a version ubuntu supports?!

**Comment by anonimal** [comment]
Hi @grummerd, thanks for opening this issue.

As stated in https://github.com/monero-project/kovri/blob/master/doc/BUILDING.md#step-3-build
## Step 3. Build
Minimum requirement:
```bash
$ git clone --recursive https://github.com/monero-project/kovri
$ make dependencies && make && make install-resources # to decrease build-time, run make -j [available CPU cores]
```
- End-users MUST run ```make dependencies``` and ```make install-resources``` for new installations

In other words:

*Minimum requirement* means you **must** run ```make dependencies``` before any other ```make``` option. You can then replace a simple ```make``` with ```make all-options``` if you'd like.

So, the full command would be like in the documentation but with ```make all-options```:
```bash
$ make dependencies && make all-options && make install-resources
```

Kovri requires the cryptopp and cpp-netlib versions that are included as submodules. Also, cpp-netlib branch 0.13-release is not an official release yet so you won't see it in any package manager. By the time we officially release, 0.13-release should be officially released by cpp-netlib. If not, there are ways to cope but that is for the far future. The same goes for cryptopp.

What we *can* do for now is offer more clarity to the build instructions or we can include ```dependencies``` into ```all-options``` (including ```dependencies``` into ```all-options``` may be overkill).

I'm open to any thoughts.

**Comment by grummerd** [comment]
@anonimal This clarifies the build process alot. Hadn't a clue that `make all-options` could be run 2nd.

> So, the full command would be like in the documentation but with make all-options

This is the info people need to know! Please add this build clarification to the current readme.md, the two additional dependencies libcrypto++dev and libcppnetlib-dev, and the notes on cpp-netlib versions

Another suggestion, mention command, `lscpu`. Maybe not everyone will know how to find out how many (virtual) CPUs they have

**Comment by anonimal** [comment]
>mention command, lscpu

```lscpu``` is not a default program on msys2/mingw or osx.  ```/proc/cpuinfo``` only applies to linux/msys2 so osx users need to use ```sysctl -n hw.ncpu```. Since building is usually geared toward developers, I imagine a simple search on how to find the number of cores for their system is out of scope for kovri, so maybe we should not clutter our documentation.

>the two additional dependencies libcrypto++dev and libcppnetlib-dev, and the notes on cpp-netlib versions

I'm reluctant to say anything about either packaged dependencies other than saying **do not install the packaged dependencies**. Is this really necessary? The needed packaged dependencies are clearly laid out and nowhere does it say to install these packages but, if it helps...

>Hadn't a clue that make all-options could be run 2nd.

I do think adding more instructions about make options and their order of operations will be helpful. It also may be easier to simply include ```dependencies``` into ```all-options```.

In all, you're welcome to fork and submit a pull-request.

**Comment by anonimal** [comment]
@grummerd new instructions + documentation has been merged.

tl;dr you don't have to run ```make dependencies``` before ```make``` anymore. Simply run ```make``` or any make option you like.
